### DISCUSSION TITLE
Issue running prisma in IPython (Dataspell IDE)

### DISCUSSION THREAD

**Discussion by VenkatTeja** [RobertCraigie/prisma-client-py#470] [Q&A]
Hi, am facing issue running prisma on IPython (Jupyter) using DataSpell IDE.
I have successfully installed prisma, ran prisma generate. but while trying to import Prisma, getting error that prisma generate is not run (screenshot attached)
However, its working if I run the same using terminal using the same venv as the IDE.
Any help?

![image](https://user-images.githubusercontent.com/12660045/187016467-bf52f8d2-b09c-493d-a1a2-1b0040d9c0e1.png)

**Comment by RobertCraigie** [comment]
Hi @VenkatTeja, sorry to hear this isn't working for you. Admittedly I haven't tested Prisma Client Python with IPython yet.

Thanks for providing the screenshot!

What does this code output for you in IPython?

```py
import prisma
print(prisma.__file__)
```

**Reply by VenkatTeja** [reply]
`D:\Neev\returns-analysis\venv\lib\site-packages\prisma\__init__.py`
Its pointing to my virtual env.

**Reply by RobertCraigie** [reply]
Hey @VenkatTeja, sorry for the late reply, I missed your reply.

Then it looks like the issue is that when you're running `prisma generate` your virtualenv is not active and when you're importing it, your virtualenv is active so it doesn't know about the generated code.

Unfortunately I cannot provide a solution as I am unfamiliar with IPython but if you can find a way to active your virtual environment when running the bash command then it should work.

**Comment by ezorita** [comment]
I had the same issue and it turned out to be `__pycache__` file permissions, which were generated from a Docker container. @RobertCraigie I believe generate uses a copytree somewhere and crashes when going over `root` owned files, should `__pycache__` folders be ignored explicitly? (this is likely a different issue).

Great job at porting Prisma to python!

**Reply by RobertCraigie** [reply]
Thanks @ezorita! In the reported issue `prisma generate` seems to run successfully, in your case did it crash?

I've pushed a fix that disables `__pycache__` folders and `*.pyc` files from being copied during generation.
